src/main/java/address/util/Range.java

```diff
@@ -0,0 +1,29 @@
+package address.util;
+
+/**
+ * Represents a range of circular positive number.
+ */
+public class Range {
+
+    private final int start;
+    private final int end;
+
+    public Range(int start, int end) {
+        this.start = start;
+        this.end = end;
+    }
+
+    /**
+     * Precondition: val >=0
+     * @param val
+     * @return
+     */
+    public boolean isWithinRange(int val){
+
+        if (end > start) {
+            return val <= end && val >= start;
+        } else {
+            return val <= end || val >= start;
+        }
```

Comment:
What if `end == start`? In this implementation, it will cover everything. Should the range just be 1 number?